pkg/csi/service/common/util.go

```diff
@@ -86,6 +86,25 @@ func GetVCenters(ctx context.Context, managers *Managers) ([]*cnsvsphere.Virtual
 	return vcenters, nil
 }
 
+// GetVCenterFromVCHost returns VirtualCenter object from specified VC host.
+// Before returning VirtualCenter objects, vcenter connection is established if
+// session doesn't exist.
+func GetVCenterFromVCHost(ctx context.Context, vCenterManager cnsvsphere.VirtualCenterManager,
```

Comment:
Can  you use function `func GetVirtualCenterInstanceForVCenterHost(ctx context.Context, vcHost string) (*VirtualCenter, error) { `@shalini-b introduced in this PR - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2043/files

Reply:
`GetVirtualCenterInstanceForVCenterHost()` gets the vCenter from `vCenterInstances` map maintained.
I see `vCenterInstances` map being initialized as a part of `GetVirtualCenterInstanceForVCenterConfig()` in metadata syncer only, so it may not provide required functionality to CSI controller

Reply:
`GetVirtualCenterInstanceForVCenterHost() `gets the vCenter from `vCenterInstances` map.
I see `vCenterInstances` map is initialized in `GetVirtualCenterInstanceForVCenterConfig()` in metadata syncer only, so may not be able to access it in CSI controller, is that possible?

Reply:
we can de-dupe the code in the next PR.